oarepo_communities/ui/communities_components/semantic-ui/js/communities_components/components/CommunityInvitationsModal/CommunityInvitationsModal.jsx

```diff
@@ -0,0 +1,221 @@
+    valuesCopy.members = serializeMembers(valuesCopy.emails.validEmails);
+    const serializedValues = serializer.serialize(valuesCopy);
+    setSubmitting(true);
+    try {
+      const response = await http.post(
+        community.links.invitations,
+        serializedValues
+      );
+      if (response.status === 204) {
+        resetForm();
+        setSuccessMessage(i18next.t("Invitations sent successfully."));
+      }
+    } catch (error) {
+      if (error.response.status >= 400) {
+        setHttpError(
+          i18next.t(
+            "The invitations could not be sent. Please try again later."
+          )
+        );
+      }
+    } finally {
+      setSubmitting(false);
+    }
+  };
+
+  const debouncedValidateEmails = useCallback(
+    _debounce((value, setFieldValue) => {
+      const emails = findAndValidateEmails(value);
+      setFieldValue("emails", emails);
+    }, 1000),
+    []
+  );
+
+  const handleChange = (value, setFieldValue) => {
+    setFieldValue("membersEmails", value);
+    debouncedValidateEmails(value, setFieldValue);
+  };
+  const usersCanInvite = rolesCanInvite.user;
+  return (
+    <Formik
+      onSubmit={onSubmit}
+      initialValues={{
+        membersEmails: "",
+        members: [],
+        message: "",
+        role: "member",
+        emails: { validEmails: [], invalidEmails: [] },
+      }}
+      validateOnChange={false}
+      validateOnBlur={true}
+      enableReinitialize={true}
+    >
+      {({ values, setFieldValue, handleSubmit, resetForm, isSubmitting }) => {
+        const validEmailsCount = values.emails.validEmails.length;
+        const invalidEmailsCount = values.emails.invalidEmails.length;
+        return (
+          <Modal
+            className="form-modal community-invitations"
+            closeIcon
+            open={isOpen}
+            onClose={close}
+            onOpen={open}
+            trigger={
+              <Button
+                className="fluid-responsive"
+                content={i18next.t("Invite...")}
+                positive
+                fluid
+                compact
+                icon="user plus"
+                labelPosition="left"
+                aria-expanded={isOpen}
+                aria-haspopup="dialog"
+              />
+            }
+          >
+            <Modal.Header>
+              {i18next.t("Invite users to the {{communityTitle}} community", {
+                communityTitle: community.metadata.title,
+              })}
+            </Modal.Header>
+            <Modal.Content>
+              <Form>
+                <Form.Field>
+                  <TextAreaField
+                    fieldPath="membersEmails"
+                    required
+                    autoFocus
+                    label={
+                      <FieldLabel
+                        label={i18next.t("Members")}
+                        icon={
+                          validEmailsCount + invalidEmailsCount > 0 &&
```

Comment:
valid > 0 && invalid === 0 would be more understandable :)

Reply:
@mirekys agreed :)